### PR TITLE
[NPM-2914] Log user and group on system-probe startup

### PR DESCRIPTION
**Pull request by tblake333**
### What does this PR do?
Adds logs to log user id, user name, group id, and group name on startup.

### Motivation
Recent support tickets have been difficult to verify whether system-probe is indeed running on `root`.

### Describe how to test/QA your changes
The logs should appear right after system-probe has started. Here is an example of logs directly after starting system-probe:
```
2023-08-16 08:20:08 PDT | SYS-PROBE | WARN | (pkg/util/log/log.go:659 in func1) | configuration key `system_probe_config.enabled` is deprecated, use `network_config.enabled` instead
2023-08-16 08:20:08 PDT | SYS-PROBE | INFO | (pkg/config/environment_detection.go:123 in detectFeatures) | 2 Features detected from environment: docker,containerd
2023-08-16 08:20:08 PDT | SYS-PROBE | INFO | (pkg/config/process.go:215 in loadProcessTransforms) | process_config.enabled is deprecated, use process_config.container_collection.enabled and process_config.process_collection.enabled instead, see https://docs.datadoghq.com/infrastructure/process#installation for more information
2023-08-16 08:20:08 PDT | SYS-PROBE | INFO | (pkg/runtime/runtime.go:27 in func1) | runtime: final GOMAXPROCS value is: 4
2023-08-16 08:20:08 PDT | SYS-PROBE | INFO | (cmd/system-probe/subcommands/run/command.go:174 in startSystemProbe) | starting system-probe v7.48.0-rc.0+git.184.34a70e7
2023-08-16 08:20:08 PDT | SYS-PROBE | INFO | (cmd/system-probe/subcommands/run/command.go:184 in startSystemProbe) | current user id/name: 0/root
2023-08-16 08:20:08 PDT | SYS-PROBE | INFO | (cmd/system-probe/subcommands/run/command.go:190 in startSystemProbe) | current group id/name: 0/root,
2023-08-16 08:20:08 PDT | SYS-PROBE | INFO | (cmd/system-probe/api/module/loader.go:55 in Register) | module ebpf disabled
2023-08-16 08:20:08 PDT | SYS-PROBE | INFO | (pkg/network/tracer/utils_linux.go:27 in IsTracerSupportedByOS) | running on platform: ubuntu
```

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: cmd/system-probe/subcommands/run/command.go [hunk 1]
```diff
@@ -171,6 +173,26 @@ func startSystemProbe(cliParams *cliParams, log log.Component, telemetry telemet
...
+		} else {
+			log.Warn("unable to resolve user")
+		}
+		groupName, err := resolver.ResolveGroup(gid)
+		if err == nil {
+			log.Infof("current group id/name: %s/%s, ", strconv.Itoa(gid), groupName)
+		} else {
+			log.Warn("unable to resolve group")
+		}
+	}
```
Comment (leeavital) [inline cmd/system-probe/subcommands/run/command.go:194]: Let's split this into a dedicated function

**Review by leeavital — APPROVED** [review]

File: cmd/system-probe/subcommands/run/command.go [hunk 2]
```diff
@@ -295,3 +298,25 @@ func setupInternalProfiling(cfg ddconfig.ConfigReader, configPrefix string, log
...
 }
+
+func logUserAndGroupID(log log.Component) {
+	resolver, err := usergroup.NewResolver()
+	if err != nil {
+		log.Warn("cannot create user/group resolver")
+	} else {
+		uid := os.Getuid()
+		gid := os.Getgid()
+		userName, err := resolver.ResolveUser(uid)
```
Comment (paulcacheux) [inline cmd/system-probe/subcommands/run/command.go:309]: You should not use the resolver from the security side here. It only exists on linux for now, and this is why your PR is failing the linter job on windows

Comment (paulcacheux) [reply cmd/system-probe/subcommands/run/command.go:309]: you should most likely use https://pkg.go.dev/os/user#Current instead here

**Review by usamasaqib — APPROVED** [review]

File: cmd/system-probe/subcommands/run/command.go [hunk 3]
```diff
@@ -295,3 +297,19 @@ func setupInternalProfiling(cfg ddconfig.ConfigReader, configPrefix string, log
 func isValidPort(port int) bool {
 	return port > 0 && port < 65536
 }
+
+func logUserAndGroupID(log log.Component) {
+	currentUser, err := user.Current()
+	if err != nil {
+		log.Warn("error fetching current user")
```
Comment (brycekahle) [inline cmd/system-probe/subcommands/run/command.go:304]: I think you need a `return` here.

Comment (brycekahle) [reply cmd/system-probe/subcommands/run/command.go:304]: You should also include the error in the log message

File: cmd/system-probe/subcommands/run/command.go [hunk 4]
```diff
@@ -295,3 +297,19 @@ func setupInternalProfiling(cfg ddconfig.ConfigReader, configPrefix string, log
...
+	}
+	uid := currentUser.Uid
+	gid := currentUser.Gid
+	log.Infof("current user id/name: %s/%s", uid, currentUser.Name)
+	currentGroup, err := user.LookupGroupId(gid)
+	if err == nil {
+		log.Infof("current group id/name: %s/%s, ", gid, currentGroup.Name)
+	} else {
+		log.Warn("unable to resolve group")
```
Comment (brycekahle) [inline cmd/system-probe/subcommands/run/command.go:313]: Please include the error in log message

File: cmd/system-probe/subcommands/run/command.go [hunk 5]
```diff
@@ -295,3 +297,20 @@ func setupInternalProfiling(cfg ddconfig.ConfigReader, configPrefix string, log
...
+	if err != nil {
+		log.Warnf("error fetching current user: %s", err)
+		return
+	}
+	uid := currentUser.Uid
+	gid := currentUser.Gid
+	log.Infof("current user id/name: %s/%s", uid, currentUser.Name)
+	currentGroup, err := user.LookupGroupId(gid)
+	if err == nil {
+		log.Infof("current group id/name: %s/%s, ", gid, currentGroup.Name)
```
Comment (brycekahle) [inline cmd/system-probe/subcommands/run/command.go:312]:
```suggestion
		log.Infof("current group id/name: %s/%s", gid, currentGroup.Name)
```

**Pull request merged**